lightning/src/ln/channel.rs

```diff
@@ -10912,13 +10917,19 @@ mod tests {
 		macro_rules! test_commitment {
 			( $counterparty_sig_hex: expr, $sig_hex: expr, $tx_hex: expr, $($remain:tt)* ) => {
 				chan.context.channel_transaction_parameters.channel_type_features = ChannelTypeFeatures::only_static_remote_key();
+				let mut holder_signer = keys_provider.derive_channel_signer(chan.context.channel_value_satoshis, chan.context.channel_keys_id);
+				holder_signer.provide_channel_parameters(&chan.context.channel_transaction_parameters);
+				chan.context.holder_signer = ChannelSignerType::Ecdsa(holder_signer);
 				test_commitment_common!($counterparty_sig_hex, $sig_hex, $tx_hex, &ChannelTypeFeatures::only_static_remote_key(), $($remain)*);
 			};
 		}
 
 		macro_rules! test_commitment_with_anchors {
 			( $counterparty_sig_hex: expr, $sig_hex: expr, $tx_hex: expr, $($remain:tt)* ) => {
 				chan.context.channel_transaction_parameters.channel_type_features = ChannelTypeFeatures::anchors_zero_htlc_fee_and_dependencies();
+				let mut holder_signer = keys_provider.derive_channel_signer(chan.context.channel_value_satoshis, chan.context.channel_keys_id);
+				holder_signer.provide_channel_parameters(&chan.context.channel_transaction_parameters);
+				chan.context.holder_signer = ChannelSignerType::Ecdsa(holder_signer);
```

Comment:
The `ChannelContext` `holder_signer` now reads its `ChannelTransactionParameters` to produce the appropriate `to_remote` scriptpubkey each time that `ChannelContext` builds a commitment transaction.

So a change in the `channel_type_features` needs to be passed to the `holder_signer` as well - but didn't see a good way to do it, so here I recreate a fresh signer, provide it the updated parameters, and assign it to `holder_signer`.